python/ray/rllib/evaluation/metrics.py

```diff
@@ -2,32 +2,49 @@
 from __future__ import division
 from __future__ import print_function
 
+import logging
 import numpy as np
 import collections
 
 import ray
 from ray.rllib.evaluation.sample_batch import DEFAULT_POLICY_ID
 
+logger = logging.getLogger(__name__)
 
-def collect_metrics(local_evaluator, remote_evaluators=[]):
+
+def collect_metrics(local_evaluator, remote_evaluators=[],
+                    timeout_seconds=180):
     """Gathers episode metrics from PolicyEvaluator instances."""
 
-    episodes = collect_episodes(local_evaluator, remote_evaluators)
-    return summarize_episodes(episodes, episodes)
+    episodes, num_dropped = collect_episodes(
+        local_evaluator, remote_evaluators, timeout_seconds=timeout_seconds)
+    metrics = summarize_episodes(episodes, episodes)
+    metrics.update(num_metric_batches_dropped=num_dropped)
```

Comment:
Can we move the logic for the warnings and stuff into `summarize_episodes(episodes, episodes, num_dropped)`? That way, it will work for non-DQN algorithms as well.